ios/Exponent/Versioned/Core/UniversalModules/Permissions/EXScopedPermissions.m

```diff
@@ -82,10 +85,13 @@ - (void)askForPermissionUsingRequesterClass:(Class)requesterClass
     UIAlertAction *allowAction = [UIAlertAction actionWithTitle:@"Allow" style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
       UM_ENSURE_STRONGIFY(self);
       NSMutableDictionary *permission = [globalPermissions mutableCopy];
-      // try to save scoped permissions - if fails than permission is denied
-      if (![self.permissionsService savePermission:permission ofType:permissionType forExperience:self.experienceId]) {
-        permission[@"status"] = [[self class] permissionStringForStatus:UMPermissionStatusDenied];
-        permission[@"granted"] = @(NO);
+      // if permission should be scoped
+      if ([self shouldVerifyScopedPermission:permissionType]) {
```

Comment:
I believe that this `if` is not needed. The `shouldVerifyScopedPermission` check if the permission is scoped. So, this code won't trigger for `notification's permission`.